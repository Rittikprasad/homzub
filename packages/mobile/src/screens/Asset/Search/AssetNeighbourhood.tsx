/* eslint-disable react/no-unused-state */
import React from 'react';
import { StyleSheet, View } from 'react-native';
import { connect } from 'react-redux';
import MapView, { Marker, PROVIDER_GOOGLE } from 'react-native-maps';
import { WithTranslation, withTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { GooglePlacesService } from '@homzhub/common/src/services/GooglePlaces/GooglePlacesService';
import { theme } from '@homzhub/common/src/styles/theme';
import Icon, { icons } from '@homzhub/common/src/assets/icon';
import { AssetSelectors } from '@homzhub/common/src/modules/asset/selectors';
import { UserSelector } from '@homzhub/common/src/modules/user/selectors';
import { CustomMarker } from '@homzhub/common/src/components/atoms/CustomMarker';
import { SelectionPicker } from '@homzhub/common/src/components/atoms/SelectionPicker';
import { BottomSheet } from '@homzhub/common/src/components/molecules/BottomSheet';
import ExploreSections from '@homzhub/mobile/src/components/molecules/ExploreSections';
import { Header } from '@homzhub/mobile/src/components/molecules/Header';
import { Asset } from '@homzhub/common/src/domain/models/Asset';
import { NeighborhoodTabs, PLACES_DATA } from '@homzhub/common/src/constants/AssetNeighbourhood';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';
import { MetricSystems } from '@homzhub/common/src/domain/models/UserPreferences';
import { PlaceTypes } from '@homzhub/common/src/services/GooglePlaces/constants';
import { IState } from '@homzhub/common/src/modules/interfaces';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/SearchStack';
import { PointOfInterest } from '@homzhub/common/src/services/GooglePlaces/interfaces';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

const EXPANDED = theme.viewport.height * 0.45;

// INTERFACES & TYPES
interface IStateProps {
  asset: Asset | null;
  metricSystem: MetricSystems;
}
interface IOwnState {
  selectedTab: NeighborhoodTabs;
  selectedPlaceType: PlaceTypes;
  isBottomSheetVisible: boolean;
  pointsOfInterest: PointOfInterest[];
  selectedPlaceId: string;
  isApiActive: boolean;
}

type libraryProps = WithTranslation & NavigationScreenProps<SearchStackParamList, ScreensKeys.AssetNeighbourhood>;
type Props = IStateProps & libraryProps;

export class AssetNeighbourhood extends React.Component<Props, IOwnState> {
  private mapRef: MapView | null = null;

  public state = {
    isBottomSheetVisible: false,
    selectedTab: NeighborhoodTabs.Nearby,
    selectedPlaceType: Object.values(PLACES_DATA[NeighborhoodTabs.Nearby])[0].key,
    pointsOfInterest: [],
    selectedPlaceId: '',
    isApiActive: false,
  };

  public componentDidMount = async (): Promise<void> => {
    await this.fetchPOIs();
    this.setState({ isBottomSheetVisible: true });
  };

  public shouldComponentUpdate = (nextProps: Props, nextState: IOwnState): boolean => {
    const { isApiActive, selectedPlaceType, selectedTab } = nextState;
    const { selectedPlaceType: oldType, selectedTab: oldTab } = this.state;
    return !(isApiActive || selectedTab !== oldTab || selectedPlaceType !== oldType);
  };

  public render = (): React.ReactNode => {
    const { asset, t, metricSystem } = this.props;
    const { selectedTab, selectedPlaceType, isBottomSheetVisible, pointsOfInterest, selectedPlaceId } = this.state;

    if (!asset) return null;
    const {
      assetLocation: { longitude, latitude },
    } = asset;
    return (
      <View style={styles.flexOne}>
        <Header type="secondary" icon={icons.leftArrow} onIconPress={this.onBackPress} title={asset.projectName} />
        <MapView
          ref={(mapRef: MapView): void => {
            this.mapRef = mapRef;
          }}
          provider={PROVIDER_GOOGLE}
          style={styles.flexOne}
          initialRegion={{
            latitude,
            longitude,
            latitudeDelta: 0.1,
            longitudeDelta: 0.1,
          }}
        >
          <Marker coordinate={{ latitude, longitude }}>
            <CustomMarker selected />
          </Marker>
          {this.renderMarkers()}
        </MapView>
        <View style={styles.floatingTab}>
          <SelectionPicker
            data={[
              { title: t('nearby'), value: NeighborhoodTabs.Nearby },
              { title: t('commute'), value: NeighborhoodTabs.Commute },
            ]}
            selectedItem={[selectedTab]}
            onValueChange={this.onTabChange}
          />
        </View>
        <BottomSheet
          visible={isBottomSheetVisible}
          renderHeader={false}
          sheetHeight={EXPANDED}
          onCloseSheet={this.onBottomSheetClose}
        >
          <ExploreSections
            placeTypes={Object.values(PLACES_DATA[selectedTab])}
            // @ts-ignore
            selectedPlaceType={PLACES_DATA[selectedTab][selectedPlaceType]}
            onPlaceTypePress={this.onPlaceTypeChange}
            selectedPoiId={selectedPlaceId}
            pointsOfInterest={pointsOfInterest}
            onPoiPress={this.onPoiPress}
            metricSystem={metricSystem}
          />
        </BottomSheet>
      </View>
    );
  };

  private renderMarkers = (): React.ReactNode => {
    const { pointsOfInterest, selectedPlaceId, selectedTab, selectedPlaceType } = this.state;
    // @ts-ignore
    const placeType = PLACES_DATA[selectedTab][selectedPlaceType];

    return pointsOfInterest.map((point: PointOfInterest) => (
      <Marker key={point.placeId} coordinate={{ latitude: point.latitude, longitude: point.longitude }}>
        <Icon
          name={placeType.mapMarker}
          color={selectedPlaceId === point.placeId ? theme.colors.active : theme.colors.darkTint5}
          size={24}
        />
      </Marker>
    ));
  };

  private onTabChange = (selectedTab: NeighborhoodTabs): void => {
    const { selectedTab: oldTab, selectedPlaceType } = this.state;

    let placeTypeToUpdate = selectedPlaceType;
    if (oldTab !== selectedTab) {
      placeTypeToUpdate = Object.values(PLACES_DATA[selectedTab])[0].key;
    }

    this.setState({ selectedTab, selectedPlaceType: placeTypeToUpdate, isBottomSheetVisible: true }, () => {
      this.fetchPOIs().then();
    });
  };

  private onPlaceTypeChange = (selectedPlaceType: PlaceTypes): void => {
    this.setState({ selectedPlaceType }, () => {
      this.fetchPOIs().then();
    });
  };

  private onBottomSheetClose = (): void => {
    this.setState({ isBottomSheetVisible: false });
  };

  private onPoiPress = (poi: PointOfInterest): void => {
    const { latitude, longitude, placeId } = poi;
    this.setState({ selectedPlaceId: placeId });
    this.mapRef?.animateCamera({
      center: {
        latitude,
        longitude,
      },
    });
  };

  private onBackPress = (): void => {
    const {
      navigation: { goBack },
    } = this.props;
    goBack();
  };

  private fetchPOIs = async (): Promise<void> => {
    const { selectedPlaceType } = this.state;
    const { asset, metricSystem } = this.props;

    if (!asset) return;
    this.setState({ isApiActive: true });
    const { assetLocation } = asset;

    try {
      const pointsOfInterest = await GooglePlacesService.getPOIs(
        assetLocation,
        selectedPlaceType,
        undefined,
        metricSystem
      );

      this.setState({ pointsOfInterest, isApiActive: false });
    } catch (e) {
      this.setState({ isBottomSheetVisible: false, isApiActive: false });
      AlertHelper.error({ message: e.message });
    }
  };
}

const styles = StyleSheet.create({
  flexOne: {
    flex: 1,
  },
  floatingTab: {
    position: 'absolute',
    alignItems: 'center',
    left: 0,
    right: 0,
    marginHorizontal: 12,
    top: 120,
  },
});

const mapStateToProps = (state: IState): IStateProps => {
  const { getAsset } = AssetSelectors;
  const { getMetricSystem } = UserSelector;

  return {
    asset: getAsset(state),
    metricSystem: getMetricSystem(state),
  };
};

export default connect(
  mapStateToProps,
  null
)(withTranslation(LocaleConstants.namespacesKey.assetDescription)(AssetNeighbourhood));
