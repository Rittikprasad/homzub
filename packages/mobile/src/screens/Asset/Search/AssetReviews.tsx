import React, { memo, useState, useEffect, useRef } from 'react';
import { StyleSheet, View } from 'react-native';
import { useSelector } from 'react-redux';
import { useTranslation } from 'react-i18next';
import { AlertHelper } from '@homzhub/common/src/utils/AlertHelper';
import { AssetRepository } from '@homzhub/common/src/domain/repositories/AssetRepository';
import { theme } from '@homzhub/common/src/styles/theme';
import { AssetSelectors } from '@homzhub/common/src/modules/asset/selectors';
import { Dropdown } from '@homzhub/common/src/components/atoms/Dropdown';
import { Label } from '@homzhub/common/src/components/atoms/Text';
import { AssetReviewCard } from '@homzhub/mobile/src/components/molecules/AssetReviewCard';
import { AssetReviewsSummary } from '@homzhub/mobile/src/components/molecules/AssetReviewsSummary';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import { AssetReview } from '@homzhub/common/src/domain/models/AssetReview';
import { SearchStackParamList } from '@homzhub/mobile/src/navigation/SearchStack';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';
import { LocaleConstants } from '@homzhub/common/src/services/Localization/constants';

const SLIDER_WIDTH = theme.viewport.width - theme.layout.screenPadding * 2;
enum SortType {
  MostRecent = 'MostRecent',
  Positive = 'Positive',
  Critical = 'Critical',
}
type Props = NavigationScreenProps<SearchStackParamList, ScreensKeys.AssetReviews>;

const AssetReviews = (props: Props): React.ReactElement => {
  const {
    navigation,
    route: { params },
  } = props;
  const reviewSummary = useSelector(AssetSelectors.getAssetReviews);
  const { t } = useTranslation(LocaleConstants.namespacesKey.assetDescription);

  const [reviews, setReviews] = useState<AssetReview[]>([]);
  const [sortType, setSortType] = useState<SortType>(SortType.MostRecent);
  const [loading, setLoading] = useState(true);
  const dropDownOptions = useRef([
    { value: SortType.MostRecent, label: t('mostRecent') },
    { value: SortType.Positive, label: t('positive') },
    { value: SortType.Critical, label: t('critical') },
  ]);

  useEffect(() => {
    try {
      AssetRepository.getListingReviews(params).then((response) => {
        setReviews(response.sort((a, b) => new Date(b.modifiedAt).getTime() - new Date(a.modifiedAt).getTime()));
      });
    } catch (err) {
      AlertHelper.error({ message: t('common:genericErrorMessage') });
    } finally {
      setLoading(false);
    }
  }, []);

  useEffect(() => {
    if (reviews.length <= 0) {
      return;
    }

    let sorted: AssetReview[] = [];

    switch (sortType) {
      case SortType.Positive:
        sorted = [...reviews].sort((a, b) => b.rating - a.rating);
        break;
      case SortType.Critical:
        sorted = [...reviews].sort((a, b) => a.rating - b.rating);
        break;
      default:
        sorted = [...reviews].sort((a, b) => new Date(b.modifiedAt).getTime() - new Date(a.modifiedAt).getTime());
        break;
    }

    setReviews(sorted);
  }, [sortType]);

  return (
    <Screen
      backgroundColor={theme.colors.white}
      headerProps={{
        type: 'secondary',
        title: t('customerReviews'),
        onIconPress: navigation.goBack,
      }}
      contentContainerStyle={styles.screen}
      isLoading={loading}
    >
      {reviewSummary && (
        <AssetReviewsSummary reviewSummary={reviewSummary} containerStyle={styles.padding} sliderWidth={SLIDER_WIDTH} />
      )}
      <View style={styles.heading}>
        <Label type="large" textType="semiBold" style={styles.reviewHeading}>
          {t('property:totalReviews', { total: reviews.length })}
        </Label>
        <Dropdown
          data={dropDownOptions.current}
          value={sortType}
          onDonePress={setSortType}
          listHeight={theme.viewport.height / 2}
          testID="drpTimeRange"
          isOutline
          containerStyle={styles.dropdownStyle}
          fontSize="regular"
        />
      </View>
      {reviews.map((review: AssetReview) => (
        <AssetReviewCard key={review.id} review={review} hideButtons />
      ))}
    </Screen>
  );
};

const memoizedComponent = memo(AssetReviews);
export { memoizedComponent as AssetReviews };

const styles = StyleSheet.create({
  screen: {
    paddingHorizontal: 0,
  },
  padding: {
    marginHorizontal: 16,
  },
  heading: {
    marginHorizontal: 16,
    marginVertical: 20,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  reviewHeading: {
    color: theme.colors.darkTint3,
  },
  dropdownStyle: {
    width: 132,
  },
});
