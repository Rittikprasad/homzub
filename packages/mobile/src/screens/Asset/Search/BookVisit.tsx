import React, { Component } from 'react';
import { withTranslation, WithTranslation } from 'react-i18next';
import { StyleSheet } from 'react-native';
import { connect } from 'react-redux';
import { bindActionCreators, Dispatch } from 'redux';
import { CommonActions } from '@homzhub/common/src/modules/common/actions';
import { CommonParamList } from '@homzhub/mobile/src/navigation/Common';
import { icons } from '@homzhub/common/src/assets/icon';
import { theme } from '@homzhub/common/src/styles/theme';
import { CollapsibleSection } from '@homzhub/mobile/src/components';
import { Screen } from '@homzhub/mobile/src/components/HOC/Screen';
import ScheduleVisitForm from '@homzhub/common/src/components/organisms/ScheduleVisitForm';
import { IReviewRefer } from '@homzhub/common/src/modules/common/interfaces';
import { NavigationScreenProps, ScreensKeys } from '@homzhub/mobile/src/navigation/interfaces';

interface IVisitState {
  isCollapsed: boolean;
  isLoading: boolean;
}

interface IDispatchProps {
  setReviewReferData: (payload: IReviewRefer) => void;
}

type libraryProps = WithTranslation & NavigationScreenProps<CommonParamList, ScreensKeys.BookVisit>;
type Props = libraryProps & IDispatchProps;

export class BookVisit extends Component<Props, IVisitState> {
  public state = {
    isCollapsed: true,
    isLoading: false,
  };

  public render(): React.ReactNode {
    const { isCollapsed, isLoading } = this.state;
    const {
      t,
      route: { params },
      navigation,
    } = this.props;
    return (
      <Screen
        backgroundColor={theme.colors.white}
        isLoading={isLoading}
        headerProps={{
          onIconPress: navigation.goBack,
          type: 'secondary',
          title: t('assetDescription:BookVisit'),
          icon: icons.close,
        }}
      >
        {/* @ts-ignore */}
        <ScheduleVisitForm
          isCollapsed={isCollapsed}
          isReschedule={params ? params.isReschedule : false}
          leaseListingId={params.lease_listing_id}
          saleListingId={params.sale_listing_id}
          userId={params.userId}
          renderCollapseSection={this.renderCollapsibleSection}
          setLoading={this.setLoading}
          onSubmitSuccess={this.onSuccess}
        />
      </Screen>
    );
  }

  private renderCollapsibleSection = (children: React.ReactElement, title: string): React.ReactElement => {
    const { isCollapsed } = this.state;
    return (
      <CollapsibleSection
        title={title}
        icon={icons.watch}
        titleStyle={styles.upcomingTitle}
        initialCollapsedValue={isCollapsed}
        onCollapse={this.handleSlotView}
      >
        {children}
      </CollapsibleSection>
    );
  };

  private onSuccess = (): void => {
    const { navigation, setReviewReferData } = this.props;
    navigation.goBack();
    setReviewReferData({ isReview: true });
  };

  private handleSlotView = (isCollapsed: boolean): void => {
    this.setState({ isCollapsed });
  };

  private setLoading = (isLoading: boolean): void => {
    this.setState({ isLoading });
  };
}

const mapDispatchToProps = (dispatch: Dispatch): IDispatchProps => {
  const { setReviewReferData } = CommonActions;
  return bindActionCreators({ setReviewReferData }, dispatch);
};

const reduxComp = connect(null, mapDispatchToProps)(BookVisit);
export default withTranslation()(reduxComp);

const styles = StyleSheet.create({
  upcomingTitle: {
    marginLeft: 12,
  },
});
