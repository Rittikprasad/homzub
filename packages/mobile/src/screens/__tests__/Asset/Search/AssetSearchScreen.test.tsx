import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import toJson from 'enzyme-to-json';
import { AssetSearchScreen } from '@homzhub/mobile/src/screens/Asset/Search/AssetSearchScreen';
import { FilterData, SearchFilter } from '@homzhub/common/src/mocks/FilterData';

describe.skip('Asset Search Screen', () => {
  let component: ShallowWrapper;
  let props: any;
  const mock = jest.fn();

  beforeEach(() => {
    props = {
      filterData: null,
      filters: SearchFilter,
      currencyData: [],
      priceRange: {},
      isLoading: false,
      searchLocation: {},
      properties: {},
      getFilterDetails: mock,
      setFilter: mock,
      getProperties: mock,
      getPropertiesListView: mock,
      setInitialFilters: mock,
      setInitialState: mock,
      setChangeStack: mock,
      navigation: {
        goBack: mock,
      },
    };
  });

  it('should render snapshot', () => {
    component = shallow(<AssetSearchScreen {...props} t={(key: string): string => key} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot for Map view', () => {
    component = shallow(<AssetSearchScreen {...props} t={(key: string): string => key} />);
    component.setState({ isMapView: true });
    expect(toJson(component)).toMatchSnapshot();
  });

  it('should render snapshot with filterData', () => {
    props = {
      ...props,
      filterData: FilterData,
    };
    component = shallow(<AssetSearchScreen {...props} t={(key: string): string => key} />);
    component.setState({ isMenuTrayCollapsed: true });
    expect(toJson(component)).toMatchSnapshot();
  });
});
