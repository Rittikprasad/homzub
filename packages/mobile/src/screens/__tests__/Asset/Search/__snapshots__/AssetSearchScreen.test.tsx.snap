// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Asset Search Screen should render snapshot 1`] = `
<Fragment>
  <View
    style={
      Object {
        "backgroundColor": "#F0F5F9",
        "height": 30,
      }
    }
  >
    <StatusBar
      animated={false}
      backgroundColor="#FFFFFF"
      barStyle="dark-content"
      showHideTransition="fade"
      translucent={true}
    />
  </View>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#E0E0E0",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "margin": 16,
              "marginBottom": 10,
              "padding": 8,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#667686"
            name="search"
            size={20}
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "marginStart": 15,
                "width": 220,
              }
            }
            textType="regular"
            type="small"
          />
        </View>
      </ForwardRef>
      <View
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginHorizontal": 5,
            "paddingVertical": 10,
          }
        }
      >
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="TYPE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="type"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="PRICE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="price"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="ROOMS"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="rooms"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          icon="filter"
          iconColor="#0084F8"
          iconSize={20}
          iconStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          key="MORE"
          onPress={[Function]}
          type="secondary"
        />
      </View>
    </View>
    <Divider />
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <withI18nextTranslation(PropertySearchMap)
        favIds={Array []}
        onFavorite={[Function]}
        onSelectedProperty={[Function]}
        searchLocation={Object {}}
        transaction_type={0}
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "bottom": 20,
            "padding": 7,
            "position": "absolute",
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#4C5F72",
              "textAlign": "center",
            }
          }
          textType="regular"
          type="small"
        >
          noResultsMapView
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "position": "absolute",
            "top": 266.8,
            "width": 750,
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "elevation": 3,
            "paddingHorizontal": 12,
            "paddingVertical": 8,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Label
          textType="regular"
          type="regular"
        >
          0
           
          propertiesFound
        </Label>
      </View>
      <ToggleButton
        icon="list"
        onToggle={[Function]}
        title="common:list"
      />
    </View>
  </ForwardRef(SafeAreaView)>
  <Loader
    visible={false}
  />
</Fragment>
`;

exports[`Asset Search Screen should render snapshot for Map view 1`] = `
<Fragment>
  <View
    style={
      Object {
        "backgroundColor": "#F0F5F9",
        "height": 30,
      }
    }
  >
    <StatusBar
      animated={false}
      backgroundColor="#FFFFFF"
      barStyle="dark-content"
      showHideTransition="fade"
      translucent={true}
    />
  </View>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#E0E0E0",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "margin": 16,
              "marginBottom": 10,
              "padding": 8,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#667686"
            name="search"
            size={20}
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "marginStart": 15,
                "width": 220,
              }
            }
            textType="regular"
            type="small"
          />
        </View>
      </ForwardRef>
      <View
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginHorizontal": 5,
            "paddingVertical": 10,
          }
        }
      >
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="TYPE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="type"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="PRICE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="price"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="ROOMS"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="rooms"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          icon="filter"
          iconColor="#0084F8"
          iconSize={20}
          iconStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          key="MORE"
          onPress={[Function]}
          type="secondary"
        />
      </View>
    </View>
    <Divider />
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <withI18nextTranslation(PropertySearchMap)
        favIds={Array []}
        onFavorite={[Function]}
        onSelectedProperty={[Function]}
        searchLocation={Object {}}
        transaction_type={0}
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "bottom": 20,
            "padding": 7,
            "position": "absolute",
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#4C5F72",
              "textAlign": "center",
            }
          }
          textType="regular"
          type="small"
        >
          noResultsMapView
        </Text>
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "justifyContent": "space-between",
            "paddingHorizontal": 16,
            "position": "absolute",
            "top": 266.8,
            "width": 750,
          },
          Object {
            "flexDirection": "row",
          },
        ]
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "elevation": 3,
            "paddingHorizontal": 12,
            "paddingVertical": 8,
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 1,
              "width": 0,
            },
            "shadowOpacity": 0.22,
            "shadowRadius": 2.22,
          }
        }
      >
        <Label
          textType="regular"
          type="regular"
        >
          0
           
          propertiesFound
        </Label>
      </View>
      <ToggleButton
        icon="list"
        onToggle={[Function]}
        title="common:list"
      />
    </View>
  </ForwardRef(SafeAreaView)>
  <Loader
    visible={false}
  />
</Fragment>
`;

exports[`Asset Search Screen should render snapshot with filterData 1`] = `
<Fragment>
  <View
    style={
      Object {
        "backgroundColor": "#F0F5F9",
        "height": 30,
      }
    }
  >
    <StatusBar
      animated={false}
      backgroundColor="#FFFFFF"
      barStyle="dark-content"
      showHideTransition="fade"
      translucent={true}
    />
  </View>
  <ForwardRef(SafeAreaView)
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#FFFFFF",
          "flex": 1,
          "justifyContent": "space-around",
        }
      }
    >
      <ForwardRef
        onPress={[Function]}
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "borderColor": "#E0E0E0",
              "borderRadius": 4,
              "borderWidth": 1,
              "flexDirection": "row",
              "margin": 16,
              "marginBottom": 10,
              "padding": 8,
            }
          }
        >
          <Icon
            allowFontScaling={false}
            color="#667686"
            name="search"
            size={20}
          />
          <Text
            numberOfLines={1}
            style={
              Object {
                "marginStart": 15,
                "width": 220,
              }
            }
            textType="regular"
            type="small"
          />
        </View>
      </ForwardRef>
      <View
        style={
          Object {
            "flex": 0,
            "flexDirection": "row",
            "justifyContent": "space-around",
            "marginHorizontal": 5,
            "paddingVertical": 10,
          }
        }
      >
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="TYPE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="type"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="PRICE"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="price"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          fontType="regular"
          key="ROOMS"
          onPress={[Function]}
          textSize="regular"
          textType="label"
          title="rooms"
          titleStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          type="secondary"
        />
        <Button
          containerStyle={
            Object {
              "flex": 0,
            }
          }
          icon="filter"
          iconColor="#0084F8"
          iconSize={20}
          iconStyle={
            Object {
              "marginHorizontal": 20,
              "marginVertical": 8,
            }
          }
          key="MORE"
          onPress={[Function]}
          type="secondary"
        />
      </View>
    </View>
    <Divider />
    <View
      style={
        Object {
          "flex": 4,
        }
      }
    >
      <withI18nextTranslation(PropertySearchMap)
        favIds={Array []}
        onFavorite={[Function]}
        onSelectedProperty={[Function]}
        searchLocation={Object {}}
        transaction_type={0}
      />
      <View
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "bottom": 20,
            "padding": 7,
            "position": "absolute",
            "width": 250,
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#4C5F72",
              "textAlign": "center",
            }
          }
          textType="regular"
          type="small"
        >
          noResultsMapView
        </Text>
      </View>
      <View
        style={
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 4,
            "padding": 15,
            "paddingTop": 10,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
      />
    </View>
  </ForwardRef(SafeAreaView)>
  <Loader
    visible={false}
  />
</Fragment>
`;
